pkg/etc/wavefront/wavefront-proxy/wavefront.conf.default

```diff
@@ -91,71 +68,136 @@ customSourceTags=fqdn, hostname
+#dataPrefillCutoffHours=24
+
+################################################## ADVANCED SETTINGS ###################################################
+## Number of threads that flush data to the server. If not defined in wavefront.conf it defaults to the
+## number of processors (min 4). Setting this value too large will result in sending batches that are
+## too small to the server and wasting connections. This setting is per listening port.
+#flushThreads=4
+## Max points per flush. Typically 40000.
+#pushFlushMaxPoints=40000
+## Milliseconds between flushes to the Wavefront servers. Typically 1000.
+#pushFlushInterval=1000
 
-## Default location of buffer.* files for saving failed transmission for retry.
-buffer=/var/spool/wavefront-proxy/buffer
+## Limit outbound pps rate at the proxy. Default: do not throttle
+#pushRateLimit=20000
+## Max number of burst seconds to allow when rate limiting to smooth out uneven traffic.
+## Set to 1 when doing data backfills. Default: 10
+#pushRateLimitMaxBurstSeconds=10
 
 ## Number of threads retrying failed transmissions. Defaults to the number of processors (min. 4)
 ## Buffer files are maxed out at 2G each so increasing the number of retry threads effectively governs
 ## the maximum amount of space the proxy will use to buffer points locally
 #retryThreads=4
-
-## Regex pattern (java.util.regex) that input lines must match to be accepted.
-## Input lines are checked against the pattern before the prefix is prepended.
-#whitelistRegex=^(production|stage).*
-
-## Regex pattern (java.util.regex) that input lines must NOT match to be accepted.
-## Input lines are checked against the pattern before the prefix is prepended.
-#blacklistRegex=^(qa|development|test).*
-
-## Whether to split the push batch size when the push is rejected by Wavefront due to rate limit.  Default false.
-#splitPushWhenRateLimited=false
-
+## Location of buffer.* files for saving failed transmission for retry. Default: /var/spool/wavefront-proxy/buffer
+#buffer=/var/spool/wavefront-proxy/buffer
 ## For exponential backoff when retry threads are throttled, the base (a in a^b) in seconds.  Default 2.0
 #retryBackoffBaseSeconds=2.0
-
-## Control whether metrics traffic from the proxy to the Wavefront endpoint is gzip-compressed. Default: true
-#gzipCompression=false
+## Whether to split the push batch size when the push is rejected by Wavefront due to rate limit.  Default false.
+#splitPushWhenRateLimited=false
 
 ## The following settings are used to connect to Wavefront servers through a HTTP proxy:
 #proxyHost=localhost
 #proxyPort=8080
 ## Optional: if http proxy requires authentication
 #proxyUser=proxy_user
 #proxyPassword=proxy_password
-#
-## The following setting enables SO_LINGER with the specified linger time in seconds (SO_LINGER disabled by default)
-#soLingerTime=0
-## HTTP connect timeout (in milliseconds). Default: 5s (5000)
+## Override User-Agent in request headers if http proxy is filtering traffic by User-Agent
```

Comment:
What does User-Agent mean in this context? Is this an artifact left from the time when we called the proxy the agent?

Reply:
That's how the proxy can pretend to be a browser. That header in HTTP standard is called User-Agent.